src/media/js/elements/header.js

```diff
@@ -0,0 +1,156 @@
+        focused. When the input is blurred, then the header child will toggle
+        closed.
+
+    <mkt-header-child-toggle>
+        Toggles the visibility of a <mkt-header-child>.
+
+        for - specifies the ID of the child it will toggle.
+
+    z.page.navigate
+        - Toggle all header children closed.
+*/
+define('elements/header',
+    ['core/z', 'jquery', 'underscore'],
+    function(z, $, _) {
+    'use strict';
+
+    // Set on <mkt-header-child> saying that it is visible.
+    var CHILD_VISIBLE = 'mkt-header-child--visible';
+    // Data attribute for whether the header child contains an input.
+    var CHILD_INPUT = 'data-header-child--input';
+
+    var MktHeaderElement = document.registerElement('mkt-header', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    // Move <mkt-header-child>ren out since we need to z-index
+                    // them below <mkt-header>.
+                    var headerChildren = root.headerChildren();
+                    for (var i = 0; i < headerChildren.length; i++) {
+                        root.parentNode.insertBefore(headerChildren[i],
+                                                     root.nextSibling);
+                    }
+                }
+            },
+            headerChildren: {
+                value: function() {
+                    return document.querySelectorAll('mkt-header-child');
+                }
+            },
+            hideHeaderChildren: {
+                value: function() {
+                    var headerChildren = this.headerChildren();
+                    for (var i = 0; i < headerChildren.length; i++) {
+                        headerChildren[i].toggle(false);
+                    }
+                }
+            }
+        })
+    });
+
+    var MktHeaderChildElement = document.registerElement('mkt-header-child', {
+        prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    if (root.hasAttribute(CHILD_INPUT)) {
+                        // If main element is input, then hide child on blur.
+                        root.querySelector('input').onblur = function() {
+                            setTimeout(function() {
+                                // setTimeout to queue behind toggle button.
+                                root.toggle(false);
+                            }, 50);
+                        };
+                    }
+                }
+            },
+            toggle: {
+                value: function(bool) {
+                    // Toggle visibility.
+                    var root = this;
+                    if (bool !== undefined) {
+                        if (bool) {
+                            root.classList.add(CHILD_VISIBLE);
```

Comment:
Ah, you're checking for `undefined` though so this could use toggle when `bool` isn't `undefined` right?

We do this in several places, might make sense to have a `toggleClass` helper.